Make secure_settings always pass values as string

inkel opened the pull request:
In #234 we have found that the secure settings for alert notifications must always be passed as a `map[string]string`, however for the case of `true` and `false` we were converting the values to booleans, which made Grafana HTTP API fail with a 400 Bad Request.

This change only removes the boolean conversions, so values are always passed as strings. A better fix that also involves the underlying API client would be to make this field a `map[string]string` instead of `map[string]interface`.

inkel committed:
Make secure_settings always pass values as string

Signed-off-by: Leandro López (inkel) <inkel.ar@gmail.com>

inkel commented on grafana/resource_alert_notification.go:
```diff
@@ -224,14 +224,10 @@ func makeAlertNotification(_ context.Context, d *schema.ResourceData) (*gapi.Ale
-		case ok && strVal == "true":
-			secureSettings[k] = true
-		case ok && strVal == "false":
-			secureSettings[k] = false
-		default:
-			secureSettings[k] = v
+		if !ok {
+			return nil, errors.New("secure_settings must be a map of string")
 		}
+		secureSettings[k] = strVal
 	}
```
I've tried different alternatives and `v` can always be casted into a string, no matter what we pass before (e.g. if we defined the value as a list then Terraform will identify it's an invalid value and fail before reaching this line). Because of this, we could have replaced all this code by the following and rely on Go's compiler to raise an error if it cannot cast the value to a string, however, this error is raised as a `panic`, so I'd rather have a more controlled error returned to the user.

```go
secureSettings := make(map[string]interface{})
for k, v := range d.Get("secure_settings").(map[string]interface{}) {
    secureSettings[k] = v.(string)
}
```

julienduchesne reviewed (commented):
So, this only seems to be the case for `secure_settings`, right? Because we have functional tests that have `settings` with booleans in them. Could we have functional tests for secure settings as well?

inkel commented:
Correct. I've tried passing booleans in `settings` and that works correctly, so the tests are reliable 😸 

I've tried a regression test for `secure_settings` but couldn't come up with a good one, if we have

```terraform
resource "grafana_alert_notification" "foo" {
  # …
  secure_settings = {
    "foo" = true
  }
}
```

with the changes in this PR, this won't be passed anymore as a boolean but rather as a string, so it always work. Same if we had an integer or any other basic kind of value.

Now, if we were to have something like:

```terraform
resource "grafana_alert_notification" "foo" {
  # …
  secure_settings = {
    "foo" = [true, false]
  }
}
```

then Terraform will catch this earlier than us and fail with:

```
│ Error: Incorrect attribute value type
│
│   on issue_234.tf line 11, in resource "grafana_alert_notification" "issue234":
│   11:   secure_settings = {
│   12:     "url" = [true, false]
│   13:   }
│
│ Inappropriate value for attribute "secure_settings": element "url": string required.
```

julienduchesne commented:
Yep, but I mean testing that we don't revert the code in the future.
 Adding 
```
secure_settings = {
		"foo" = "true"
	}
```
to `testAccAlertNotificationConfig_basic` makes the tests fail:

```
=== RUN   TestAccAlertNotification_basic
    resource_alert_notification_test.go:20: Step 1/1 error: Error running apply: exit status 1
        
        Error: status: 400, body: [{"classification":"DeserializationError","message":"json: cannot unmarshal bool into Go struct field CreateAlertNotificationCommand.secureSettings of type string"}]
        
          with grafana_alert_notification.test,
          on terraform_plugin_test.tf line 2, in resource "grafana_alert_notification" "test":
           2: resource "grafana_alert_notification" "test" {
```


With your fix, it doesn't fail anymore and we'll be tested for this case in the future

inkel commented:
Oh, I see what you mean. Yeah, I can do that.

inkel committed:
Add regression test for secure_settings map

By adding this in the basic test, if we ever reintroduce this bug the
test will fail.

Signed-off-by: Leandro López (inkel) <inkel.ar@gmail.com>

julienduchesne reviewed (approved):
LGTM!
